Tests: LuceneTests are failing on Java9

Seems that `LuceneTests.testMMapHackSupported` fails regularly on Java 9 right now, independent from the ES version

```
REPRODUCE WITH: gradle :core:test -Dtests.seed=ECC3BC902EFF7263 -Dtests.class=org.elasticsearch.common.lucene.LuceneTests -Dtests.method="testMMapHackSupported" -Dtests.security.manager=true -Dtests.jvm.argline="--add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.nio.file=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.locks=ALL-UNNAMED --add-opens=java.base/java.util.regex=ALL-UNNAMED" -Dtests.locale=mas-KE -Dtests.timezone=Zulu
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.1+java9-periodic/540/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.0+java9-periodic/1324/console

Hmm, this is not good: it means on Java 9 we (ES) are failing to use Lucene's unmapping best efforts.  Here's the root cause:

```
FAILURE 0.02s J0 | LuceneTests.testMMapHackSupported <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: MMapDirectory does not support unmapping: Unmapping is not supported on this platform, because internal Java APIs are not compatible to this Lucene version: java.lang.IllegalAccessException: symbolic reference class is not accessible: class jdk.internal.ref.Cleaner, from org.apache.lucene.store.MMapDirectory (unnamed module @77dd2cd6)
   > 	at __randomizedtesting.SeedInfo.seed([ECC3BC902EFF7263:1D3CDF5BF2E129A4]:0)
   > 	at org.elasticsearch.common.lucene.LuceneTests.testMMapHackSupported(LuceneTests.java:408)
```

I don't think it's a missing security permission, because Lucene throws a different message in that case.

Maybe our 1.9-ea is too old in the ES jenkins instance?  We are running with `9-ea-150` but from https://jdk9.java.net/ it looks like it's up to `ea-151`.

Otherwise I'm not sure what's wrong ... @uschindler any pointers welcome :)

--- a/core/src/test/java/org/elasticsearch/common/lucene/LuceneTests.java
+++ b/core/src/test/java/org/elasticsearch/common/lucene/LuceneTests.java
@@ -406,7 +406,7 @@ public class LuceneTests extends ESTestCase {
      */
     public void testMMapHackSupported() throws Exception {
         // add assume's here if needed for certain platforms, but we should know if it does not work.
-        assumeTrue("Lucene 6.3.0 doesn't support unmap with Java 9", Constants.JVM_SPEC_VERSION.startsWith("1.9") == false);
+        assumeTrue("Lucene 6.3.0 doesn't support unmap with Java 9", Constants.JRE_IS_MINIMUM_JAVA9 == false);
         assertTrue("MMapDirectory does not support unmapping: " + MMapDirectory.UNMAP_NOT_SUPPORTED_REASON, MMapDirectory.UNMAP_SUPPORTED);
     }
 }

--- a/core/src/test/java/org/elasticsearch/common/lucene/LuceneTests.java
+++ b/core/src/test/java/org/elasticsearch/common/lucene/LuceneTests.java
@@ -43,6 +43,7 @@ import org.apache.lucene.store.Directory;
 import org.apache.lucene.store.MMapDirectory;
 import org.apache.lucene.store.MockDirectoryWrapper;
 import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.Constants;
 import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
@@ -405,6 +406,7 @@ public class LuceneTests extends ESTestCase {
      */
     public void testMMapHackSupported() throws Exception {
         // add assume's here if needed for certain platforms, but we should know if it does not work.
+        assumeTrue("Lucene 6.3.0 doesn't support unmap with Java 9", Constants.JVM_SPEC_VERSION.startsWith("1.9") == false);
         assertTrue("MMapDirectory does not support unmapping: " + MMapDirectory.UNMAP_NOT_SUPPORTED_REASON, MMapDirectory.UNMAP_SUPPORTED);
     }
 }

